### PR TITLE
BUG - anchoring dates for resample with Day(n>1)

### LINKED ISSUE #24127: pandas.DataFrame.resample  behaves inconsistently when the rule is set to 24H/1D and 48H/2D
#### Code Sample, a copy-pastable example if possible

```python
>>> pd.__version__
'0.23.4'
>>> index = pd.date_range('11/11/2000 06:00:00', periods=50, freq='H')
>>> series = pd.Series(range(50), index=index)
>>> series.head(10)
2000-11-11 06:00:00    0
2000-11-11 07:00:00    1
2000-11-11 08:00:00    2
2000-11-11 09:00:00    3
2000-11-11 10:00:00    4
2000-11-11 11:00:00    5
2000-11-11 12:00:00    6
2000-11-11 13:00:00    7
2000-11-11 14:00:00    8
2000-11-11 15:00:00    9
Freq: H, dtype: int64
>>> series.resample('24H').count()
2000-11-11    18
2000-11-12    24
2000-11-13     8
Freq: 24H, dtype: int64
>>> series.resample('1D').count()
2000-11-11    18
2000-11-12    24
2000-11-13     8
Freq: D, dtype: int64
>>> series.resample('48H').count()
2000-11-11    42
2000-11-13     8
Freq: 48H, dtype: int64
>>> series.resample('2D').count()
2000-11-11 06:00:00    48
2000-11-13 06:00:00     2
dtype: int64

```
#### Problem description

As you can see, when I set rule to '24H' or '1D', the behavior of resample is consistent (in both cases it starts at 0 o'clock on the first day), but for '48H' and '2D' it's obviously not.

#### Expected Output
If the behavior consistency of ‘24H’ and ‘1D’ is correct, then the behaviors of '48H' and '2D' should also be consistent.
#### Output of ``pd.show_versions()``

<details>

commit: None
python: 3.6.6.final.0
python-bits: 64
OS: Windows
OS-release: 10
machine: AMD64
processor: Intel64 Family 6 Model 158 Stepping 9, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None
LOCALE: None.None
pandas: 0.23.4
pytest: None
pip: 18.1
setuptools: 40.4.3
Cython: None
numpy: 1.15.2
scipy: 1.1.0
pyarrow: None
xarray: None
IPython: None
sphinx: None
patsy: None
dateutil: 2.7.3
pytz: 2018.5
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: None
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: None
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: None
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None

</details>

### PR DESCRIPTION
**Pull request by ghost**
- [ ] closes #24127
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by ArtinSarraf** [comment]
@jreback - I don't see any open issues that seem to be directly relevant to this issue. I see a few resampling ones, but they have more to do with the behavior of the `closed` parameter. This issue is specific to resampling with pd.offsets.Day(n>1). If you can point me to any issues I can take a look and add test cases for it.

File: pandas/tests/groupby/test_timegrouper.py [hunk 1]
```diff
@@ -43,8 +43,8 @@ def test_groupby_with_timegrouper(self):
 
             expected = DataFrame(
                 {'Quantity': 0},
-                index=date_range('20130901 13:00:00',
-                                 '20131205 13:00:00', freq='5D',
+                index=date_range('20130901',
```
Comment (mroeschke) [inline pandas/tests/groupby/test_timegrouper.py]: Why did this test have to change?

File: pandas/tests/resample/test_datetime_index.py [hunk 2]
```diff
@@ -1463,3 +1463,25 @@ def f(data, add_arg):
...
         assert_frame_equal(result, expected)
+
+    @pytest.mark.parametrize('k', [1, 2, 3])
+    @pytest.mark.parametrize('n1, freq1, n2, freq2', [
+        (60, 'Min', 1, 'H'),
+        (1440, 'Min', 1, 'D'),
+        (24, 'H', 1, 'D'),
+        (60, 'S', 1, 'Min'),
+        (3600, 'S', 1, 'H'),
+        (86400, 'S', 1, 'D')
```
Comment (mroeschke) [inline pandas/tests/resample/test_datetime_index.py]: Could you also add a frequency where the n=0.5? e.g. 0.5D and 12 H

**Review by mroeschke — CHANGES_REQUESTED** [review]

File: pandas/tests/groupby/test_timegrouper.py (hunk 1 above)
Comment (mroeschke) [reply pandas/tests/groupby/test_timegrouper.py]: What was the result/expected before and the result/expected after this change?

Comment (mroeschke) [reply pandas/tests/groupby/test_timegrouper.py]: Makes sense. Thanks

**Review by mroeschke — APPROVED** [review]
LGTM. Off to you @jreback

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @ArtinSarraf 

I have a feeling this might solve some open resample issues, if you'd have a look thru the tracker (resample is a label)

**Comment by ArtinSarraf** [comment]
@jreback - I just looked through all the open issues labeled with "resample" I didn't see any other ones that I think this would have solved, since this is very specific to daily frequencies.

**Pull request merged**